compiler/damlc/lib/DA/Cli/Visual.hs

```diff
@@ -81,42 +81,49 @@ darToWorld manifest pkg = AST.initWorldSelf pkgs pkg
 tplName :: LF.Template -> T.Text
 tplName LF.Template {..} = head (LF.unTypeConName tplTypeCon)
 
-handleChoiceAndAction :: ChoiceAndAction -> LF.ChoiceName
+handleChoiceAndAction :: ChoiceAndAction -> (LF.ChoiceName, IsConsuming)
 handleChoiceAndAction (ChoiceAndAction tpl choice _)
-    | LF.chcName choice == LF.ChoiceName "Create" = LF.ChoiceName $ tplName tpl <> "_Create"
-    | LF.chcName choice == LF.ChoiceName "Archive" = LF.ChoiceName $ tplName tpl <> "_Archive"
-    | otherwise = LF.chcName choice
+    | LF.chcName choice == LF.ChoiceName "Create" = (LF.ChoiceName $ tplName tpl <> "_Create", False)
+    | LF.chcName choice == LF.ChoiceName "Archive" = (LF.ChoiceName $ tplName tpl <> "_Archive", True)
+    | otherwise = (LF.chcName choice, LF.chcConsuming choice)
 
+type IsConsuming = Bool
 -- Making choiceName is very weird
-handleCreateAndArchive :: TemplateChoiceAction -> [LF.ChoiceName]
+handleCreateAndArchive :: TemplateChoiceAction -> [(LF.ChoiceName, IsConsuming)]
 handleCreateAndArchive TemplateChoiceAction {..} = [createChoice, archiveChoice] ++ map handleChoiceAndAction choiceAndAction
-    where archiveChoice = LF.ChoiceName $ tplName template <> "_Archive"
-          createChoice = LF.ChoiceName $ tplName template <> "_Create"
+    where archiveChoice = (LF.ChoiceName $ tplName template <> "_Archive", True)
+          createChoice = (LF.ChoiceName $ tplName template <> "_Create", False)
```

Comment:
I'm finding the relationship between this function and the one for ChoiceAndAction a bit hard to understand. It seems like there is some duplication here - is that for a good reason?

Reply:
We changed this yesterday.